pkg/agent/core/state.go

```diff
@@ -199,9 +206,10 @@ func (ns *neonvmState) ongoingRequest() bool {
 func NewState(vm api.VmInfo, config Config) *State {
 	return &State{
 		internal: state{
-			Config: config,
-			Debug:  false,
-			VM:     vm,
+			Config:          config,
+			Debug:           false,
+			LoggedResources: api.Resources{VCPU: 0, Mem: 0},
```

Comment:
This is a somewhat tricky change to do well, and will intersect with #983. Let's split into a separate PR?

(broader context is that `core.State` is complicated and we do want to periodically log the desired resources even if they haven't changed (either time-based or "metrics changed"), and sometimes we'll want to immediately log them even if they're equal to what the VM is using.)

Reply:
Splitted into https://github.com/neondatabase/autoscaling/pull/1014